models/issues/issue.go

```diff
@@ -442,6 +443,30 @@ func (issue *Issue) PatchURL() string {
 	return ""
 }
 
+/* the logic should be kept the same as getIssueIcon/getIssueColor in TS code */
+func (issue *Issue) IconHTML(ctx context.Context) template.HTML {
+	if issue.IsPull {
```

Comment:
Maybe this logic can be better rewritten to switch? otherwise it's very hard to follow

Reply:
2ffcd2e0b8